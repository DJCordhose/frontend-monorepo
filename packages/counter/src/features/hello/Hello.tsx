import React from "react";
import { useTranslation } from "react-i18next";

import styled from 'styled-components';

const Div = styled.div`
  color: tomato;
  border-color: tomato;
  border-width: 0.1em;
`;

function Hello() {
  const { t } = useTranslation();

  return (
    <div id='hello-app' className="p-6 max-w-sm mx-auto space-x-5">
      <Div>
        {t("hello")}
      </Div>
    </div>
  );
}

export default Hello;
