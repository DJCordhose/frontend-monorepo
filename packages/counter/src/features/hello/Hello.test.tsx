import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { store } from "../../app/store";
import Hello from "./Hello";
import { BrowserRouter as Router } from "react-router-dom";

// https://react.i18next.com/misc/testing
jest.mock("react-i18next", () => ({
  // this mock makes sure any components using the translate hook can use it without a warning being shown
  useTranslation: () => {
    return {
      t: (str: string) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

// https://jestjs.io/docs/en/snapshot-testing
test("snapshot: renders correctly", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Hello />
        </Router>
      </Provider>
    )
    .toJSON();
    
  expect(tree).toMatchSnapshot();
});
