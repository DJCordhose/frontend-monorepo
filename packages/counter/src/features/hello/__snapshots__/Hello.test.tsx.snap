// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot: renders correctly 1`] = `
<div
  className="p-6 max-w-sm mx-auto space-x-5"
  id="hello-app"
>
  <div
    className="sc-bdnxRM epder"
  >
    hello
  </div>
</div>
`;
